lib/iiif_print/metadata.rb

```diff
@@ -40,25 +40,28 @@ def build_metadata_for_v2
       end.compact
     end
 
+    # rubocop:disable Metrics/AbcSize
     def build_metadata_for_v3
       fields.map do |field|
-        values = Array(work.try(field.name)).map { |value| scrub(value.to_s) }
-        next if values.empty?
-        {
-          'label' => {
-            # Since we're using I18n to translate the field, we're setting the locale used in the translation.
-            I18n.locale.to_s => [Hyrax::Renderers::AttributeRenderer.new(field.name, nil).label]
-          },
-          'value' => {
-            'none' => values
-          }
-        }
+        values = Array(work["#{field.name}_tesim"]).map { |value| scrub(value.to_s) }
+        if field.name == :collection && member_of_collection?
+          viewable_collections = Hyrax::CollectionMemberService.run(SolrDocument.find(work.id), @current_ability)
```

Comment:
It looks like we have a duplication of `SolrDocument.find` consider extracting a `solr_doc` method that does this logic.